src/library/mixin_helper.lua

```diff
@@ -578,4 +597,64 @@ function mixin_helper.boolean_to_error(object, method, ...)
     end
 end
 
+--[[
+% create_localized_proxy
+
+Creates a proxy method that takes localization keys instead of raw strings.
+
+@ method_name (string)
+@ class_name (string|nil) If `nil`, the resulting call will be on the `self` object. If a `string` is passed, it will be forwarded to a static call on that class in the `mixin` namespace.
+@ only_localize_args (table|nil) If `nil`, all values passed to the method will be localized. If only certain arguments need localizing, pass a `table` of argument `number`s (note that `self` is argument #1).
+: (function)
+]]
+function mixin_helper.create_localized_proxy(method_name, class_name, only_localize_args)
+    local args_to_localize
+    if only_localize_args == nil then
+        args_to_localize = setmetatable({}, { __index = function() return true end })
+    else
+        args_to_localize = utils.create_lookup_table(only_localize_args)
+    end
+
+    return function(self, ...)
+        local args = table.pack(...)
+
+        for arg_num = 1, args.n do
+            if args_to_localize[arg_num] then
+                mixin_helper.assert_argument_type(arg_num, args[arg_num], "string", "FCString")
+                args[arg_num] = localization.localize(mixin_helper.to_string(args[arg_num]))
+            end
+        end
+
+        --Tail call. Errors will pass through to the correct level
+        return (class_name and mixin[class_name] or self)[method_name](self, table.unpack(args, 1, args.n))
+    end
+end
+
+--[[
+% process_string_arguments
+
+Process multiple string arguments.
+
+@ self (class instance)
+@ method_func (function) A method on the class that accepts a single Lua `string` or `FCString` instance
+@ ... (table, FCStrings | FCString | string | number)
+]]
+function mixin_helper.process_string_arguments(self, method_func, ...)
+    for i = 1, select("#", ...) do
+        local v = select(i, ...)
+        mixin_helper.assert_argument_type(i + 1, v, "table", "string", "number", "FCString", "FCStrings")
+
+        if type(v) == "userdata" and v:ClassName() == "FCStrings" then
+            for str in each(v) do
+                method_func(self, str)
+            end
+        elseif type(v) == "table" then
+            mixin_helper.process_string_arguments(self, method_func, table.unpack(v))
```

Comment:
I had intentionally omitted the ability to pass tables as I explained here: https://github.com/finale-lua/lua-scripts/pull/183#issuecomment-1134205699